### DISCUSSION TITLE
Dịch AutoGraph và Tracing

### DISCUSSION THREAD

**Discussion by quangvnai** [mlbvn/handson-ml2-vn#139] [Terminology] [answered]
**AutoGraph**
Ta nên dịch `Đồ thị Tự động` hay giữ nguyên từ AutoGraph?

Context:
```
So how does TensorFlow generate graphs? It starts by analyzing the Python function’s source code to capture all 
the control flow statements, such as \texttt{for} loops, \texttt{while} loops, and \texttt{if} statements, .. 
After analyzing the function’s code, AutoGraph outputs an upgraded version of that function...
```

**Tracing**
Theo mình hiểu thì trong quá trình tracing, Tensorflow sẽ thực hiện tính toán dựa trên đồ thị và các giá trị thật truyền vào.
Lướt qua thấy ngành đồ họa sử dụng từ này để `dựng/vẽ`. Vẫn chưa tìm được từ dịch phù hợp cho từ này, mọi người có đề xuất gì k ạ? Có từ `truy vết` nghe cũng khá sát nữa.

Context:
```
In Figure \autoref{autograph_tracing_diagram}, 
% you can see the \texttt{tf_sum_squares()} function being called with a symbolic tensor as its argument 
% (in this case, an int32 tensor of shape []) and the final graph being generated during tracing. 
```

**Comment by ngcthuong** [comment, marked as answer]
`AutoGraph` transforms a subset of Python which operates on TensorFlow objects into equivalent TensorFlow graph code. When executing the graph, it has the same effect as if you ran the original code in eager mode. Python code which doesn't operate on TensorFlow objects remains functionally unchanged, but keep in mind that tf.function only executes such code at trace time, and generally will not be consistent with eager execution.

Theo nghĩa gốc thì không sát với từ "Đồ thị tự động", "dựng/vẽ" hay "truy vết".  Mình thiên về để nguyên hơn và giải nghĩa.

**Reply by rootonchair** [reply]
cái "dựng/vẽ" hay "truy vết" là dành cho từ "tracing" mà anh. 

Từ "Tracing" em thấy theo "truy vết" là sát nghĩa

Từ "AutoGraph" để nguyên em thấy ok. Cái này nó giống như tên của thằng TF đặt cho cơ chế này của nó

**Reply by quangvnai** [reply]
Theo anh Thướng và Vinh thì ta chốt giữ nguyên từ AutoGraph, còn Tracing thì dịch là Truy vết nhé ;)
